pilot/pkg/xds/bootstrapds.go

```diff
@@ -54,9 +56,23 @@ func (e *BootstrapGenerator) Generate(proxy *model.Proxy, _ *model.PushContext,
 	if err = jsonpb.Unmarshal(io.Reader(&buf), bs); err != nil {
 		log.Warnf("failed to unmarshal bootstrap from JSON %q: %v", buf.String(), err)
 	}
+	bs = e.applyPatches(bs, proxy, push)
 	return model.Resources{
 		&discovery.Resource{
 			Resource: util.MessageToAny(bs),
 		},
 	}, model.DefaultXdsLogDetails, nil
 }
+
+func (e *BootstrapGenerator) applyPatches(bs *bootstrapv3.Bootstrap, proxy *model.Proxy, push *model.PushContext) *bootstrapv3.Bootstrap {
+	patches := push.EnvoyFilters(proxy)
+	if patches == nil {
+		return bs
+	}
+	for _, patch := range patches.Patches[networking.EnvoyFilter_BOOTSTRAP] {
+		if patch.Operation == networking.EnvoyFilter_Patch_MERGE {
```

Comment:
Error when other operations are used, or is it already covered in validation ?

Reply:
Validation is not checking for matching operation/applyTo. Error is not an option since we have to return something. Is logging OK here?

Reply:
Can we add it to the validating webhook? a separate PR is fine. 
Logging is ok here.

Reply:
Do we have any precedent for incrementing metric for validation errors in pilot like this?

Reply:
Validation is more of a hassle than it's worth IMHO. Merge is the only operation that makes sense with bootstrap, so it's a fairly obvious mistake not to use Merge.

Reply:
Added a metric increment for malformed bootstrap patch.